kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/offer/OfferRepository.java

```diff
@@ -5,17 +5,23 @@
 import com.kin.ecosystem.Callback;
 import com.kin.ecosystem.base.ObservableData;
 import com.kin.ecosystem.exception.DataNotAvailableException;
+import com.kin.ecosystem.marketplace.model.NativeOffer;
+import com.kin.ecosystem.marketplace.model.NativeSpendOffer;
 import com.kin.ecosystem.network.model.Offer;
+import com.kin.ecosystem.network.model.Offer.ContentTypeEnum;
 import com.kin.ecosystem.network.model.OfferList;
+import java.util.Iterator;
 
 public class OfferRepository implements OfferDataSource {
 
     private static OfferRepository instance = null;
 
     private final OfferDataSource.Remote remoteData;
 
-    private OfferList cachedOfferList;
-    private ObservableData<Offer> pendingOffer = ObservableData.create();;
+    private OfferList cachedOfferList = new OfferList();
+    private ObservableData<Offer> pendingOffer = ObservableData.create();
+
+    private static Callback<NativeSpendOffer> nativeSpendOfferCallback;
```

Comment:
should nativeSpendOfferCallback be weak reference? we don't know where Digital Service developer call it. It can create an anonymous callback that hold reference to another object etc.

Reply:
Can't be a waekreference since it could be an anon inner class, i'll make it observer, where the dev can remove themselves.